app/controllers/subscribers_controller.rb

```diff
@@ -10,25 +11,42 @@ def subscriptions
   end
 
   def change_address
-    subscriber.update!(address: new_address)
+    subscriber = find_subscriber(subscriber_params.require(:address))
+    subscriber.update!(address: subscriber_params.require(:new_address))
     render json: { subscriber: subscriber }
   end
 
-private
-
-  def subscriber
-    @subscriber ||= Subscriber.find_by!("LOWER(address) = ?", address.downcase)
+  def auth_token
```

Comment:
I'm not really picturing the flow of the aut_token through the api and frontend at the moment....why do we need it as a public endpoint? is the idea to auth token to `email-alert-frontend` when we get a new subscriber and as such create the session?

Reply:
The endpoint here is only available internally and a rough sketch of the process is:

- You visit gov.uk/email/authenticate and are served a form
- You submit that form by posting an email address to /email/authenticate. This then
  - Sends a request to Email Alert API to create an auth token
  - That token is emailed to the user as part of a link
- The user is then given a message saying check email
- Then the email to login flow kicks in.

Does that help?

Reply:
👍 yep thanks